abcclassroom/template.py

```diff
@@ -197,30 +248,33 @@ def copy_assignment_files(config, template_repo, assignment):
     template_repo: os path object
         Absolute path to the template repository where the assignment files
         will be copied
-    assignment: string
+    release_dir: Path object.... EDIT ME
         name of the assignment being copied
 
     """
 
     course_dir = cf.get_config_option(config, "course_directory", True)
-    materials_dir = cf.get_config_option(config, "course_materials", True)
-    parent_path = utils.get_abspath(materials_dir, course_dir)
-    release_dir = Path(parent_path, "release", assignment)
-
-    if not release_dir.is_dir():
-        print(
-            "release directory {} does not exist; exiting\n".format(
-                release_dir
-            )
-        )
-        sys.exit(1)
+    # materials_dir = cf.get_config_option(config, "course_materials", True)
+
+    # Turn this into a try / except? - or did i capture this test above?
+    # TODO - test this because i think i implemented this as a try/ except
+    #  above so we are good to remove it here.
+    # if not release_dir.is_dir():
+    #     print(
+    #         "release directory {} does not exist; exiting\n".format(
+    #             release_dir
+    #         )
+    #     )
+    #     # TODO - talk to karen about sys.exit vs catching this because
+    #     # in Python sys.exit() locks up the console
+    #     sys.exit(1)
```

Comment:
@kcranston question for you i spent just a bit of time researching try/ except with sys.exit and sys.exit on it's own but i'm still not clear what best practices are. sys.exit(1) when run in python actually locks up the console totally. i have not figured out how to get my envt back after it runs. 
my questions

1. do you know a way to return to the console command prompt with your envt in tact after sys.exit()?
2. is there any reason why we couldn't just use traditional try/excepts and catch the exception? i just am less familiar with using `sys.exit()` so i figured i would ask.

Reply:
When you say 'console', do you mean in Jupyter? That is unfortunate. Even with try / catch, we often use sys.exit(1) after catching an error. On my local machine (zsh in Terminal), it simply exits the script and leaves my environment intact.

Reply:
@kcranston yes it's odd. i'm working in pycharm on coding. so i have a python console / interpreter that is running the dev envt . yes it really locks everything up. 
it's  fine at the terminal but not fun in vanilla python. i should try python in  bash but i would assume it's the same as python via pycharm. 

have you tried to  run things line by line in an  interperter like pycharm or spyder or something like that?

i wasn't sure if sys.exit() was to halt the command line activities vs try/ catch.

Reply:
Ok, doing some research, it seems that sys.exit (or throwing SystemExit) may kill the kernel, so that's not good.

Reply:
ok thank you for the sanity check. it's been driving me a bit batty having to rebuild my envt each time when working on things! So... is a traditional try / except approach better? that is what i'm more familiar with.  when i tried it in the template module, it printed all of the traceback and then my message